pandas/util/testing.py

```diff
@@ -2533,7 +2533,7 @@ def exception_matches(self, exc_type, exc_value, trace_back):
                         pat=self.regexp.pattern, val=val
                     )
                     e = AssertionError(msg)
-                    raise_with_traceback(e, trace_back)
+                    raise e.with_traceback(trace_back)
```

Comment:
since `assert_raises_regex` has been deprecated internally, can we just remove the complete `_AssertRaisesContextmanager` class.

Reply:
Not overly familiar with this code but seems very logical...I'll give it a shot and see what happens

Reply:
Hmm well thinking this through some more some of our `pandas.util.testing` items have leaked into the public API, so we might want to deprecate first. I'll open an issue to discuss

Reply:
It was publicly deprecated in 0.24, so with the new version policy, I _think_ ok to remove in 1.0

Reply:
Ah... missed that. Makes sense - thanks